### PR TITLE
first version

### PR DESCRIPTION
**Pull request by Phmank**

### CODE REVIEW

File: frontend/components/VariantSelect/components/BaseSelector/index.jsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  }
+
+  /**
+   * Handel selection click
+   */
+  handelClick = () => {
+    const {
+      characteristicsId, value, select, additionalAction,
+    } = this.props;
+    select({ id: characteristicsId, value: value.id });
```
Comment (patrick-scherer) [inline frontend/components/VariantSelect/components/BaseSelector/index.jsx:44]: ESLint

File: frontend/components/VariantSelect/components/BaseSelector/index.jsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  };
+
+  /**
+   * Render basic selector component
+   * @return {JSX}
+   */
+  render() {
+    const { id, selectable } = this.props.value;
+    return (
+      <button
```
Comment (patrick-scherer) [inline frontend/components/VariantSelect/components/BaseSelector/index.jsx:55]: ESLint

File: frontend/components/VariantSelect/components/ImageSwatchSelector/index.jsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+/**
+ * Color Swatch component
+ */
+class ImageSwatchSelector extends Component {
+  /**
+   * Handel click and sends color selection report
+   */
+  handelClick = () => {
+    const { characteristicsId, value } = this.props;
+    this.props.sendReport({ id: characteristicsId, value: value.id });
```
Comment (patrick-scherer) [inline frontend/components/VariantSelect/components/ImageSwatchSelector/index.jsx:20]: ESLint

File: frontend/components/VariantSelect/components/ImageSwatchSelector/index.jsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import getConfig from '../../../../helpers/getConfig';
+
+const {
+  showLableBeneathImage,
+} = getConfig();
+
+/**
+ * Color Swatch component
+ */
+class ImageSwatchSelector extends Component {
```
Comment (patrick-scherer) [inline frontend/components/VariantSelect/components/ImageSwatchSelector/index.jsx:14]: Missing props validations

Comment (patrick-scherer) [inline frontend/components/VariantSelect/components/ImageSwatchSelector/index.jsx:14]: Missing props validations

File: frontend/components/VariantSelect/index.jsx [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+        </h1>
+      ) :
+      (
+        <h1
+          className={styles.headline}
+          aria-label={ariaLabel}
+          id={headlineId}
+        >
+          {props.label}
+          :
```
Comment (patrick-scherer) [inline frontend/components/VariantSelect/index.jsx:133]: Add a non-breaking space here?

h1-no-space

File: frontend/components/VariantSelect/index.jsx [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+import I18n from '@shopgate/pwa-common/components/I18n';
+import { isImageSwatch } from './helpers';
+import TileSelector from './components/TileSelector';
+import ImageSwatchSelector from './components/ImageSwatchSelector';
+import connector from './connector';
+import styles from './styles';
+
+/**
+ * VariantSelect component
+ * @param {Object} props Component props
```
Comment (patrick-scherer) [inline frontend/components/VariantSelect/index.jsx:15]: ESlint

File: frontend/components/VariantSelect/selectors.js [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+      if (additionalProperties) {
+        property = additionalProperties.find(p => p.label === `swatchImage~${value.label}`);
+
+        if (!property) {
+          imageUrl = null;
+          imageOverlayLabel = null;
+        } else {
+          imageUrl = property.value;
+          imageOverlayLabel = property.label.split('~')[1];
```
Comment (patrick-scherer) [inline frontend/components/VariantSelect/selectors.js:68]: ESLint

File: README.md [hunk 8]
```diff
@@ -1,2 +1,42 @@
...
+  "imageCharacteristic": "Color",
+  "imageSize": 75,
+  "selectedColor": "#000",
+  "showLableBeneathImage": true,
+}
+```
+
+## Important informations
+
+This extension needs to have an "eval job" set up by the Shopgate Support.
```
Comment (patrick-scherer) [inline README.md:26]: Info about property format (`swatchImage~`)

**Review by patrick-scherer — CHANGES_REQUESTED** [review]

File: frontend/components/VariantSelect/components/BaseSelector/index.jsx (hunk 1 above)
Comment (Phmank) [reply frontend/components/VariantSelect/components/BaseSelector/index.jsx:44]: done

**Pull request merged**